src/aggregator/aggregator/aggregator.go

```diff
@@ -799,45 +809,78 @@ func (m *aggregatorAddMetricMetrics) ReportError(err error) {
 	}
 }
 
-type aggregatorAddUntimedMetrics struct {
-	aggregatorAddMetricMetrics
+type aggregatorAddUntimedErrorMetrics struct {
+	aggregatorAddMetricErrorMetrics
 
 	invalidMetricTypes tally.Counter
 }
 
+func newAggregatorAddUntimedErrorMetrics(
+	scope tally.Scope,
+) aggregatorAddUntimedErrorMetrics {
+	return aggregatorAddUntimedErrorMetrics{
+		aggregatorAddMetricErrorMetrics: newAggregatorAddMetricErrorMetrics(scope),
+		invalidMetricTypes: scope.Tagged(map[string]string{
+			"reason": "invalid-metric-types",
+		}).Counter("errors"),
+	}
+}
+
+type aggregatorAddUntimedMetrics struct {
+	aggregatorAddMetricSuccessMetrics
+
+	leaderErrors    aggregatorAddUntimedErrorMetrics
+	nonLeaderErrors aggregatorAddUntimedErrorMetrics
+}
+
 func newAggregatorAddUntimedMetrics(
 	scope tally.Scope,
 	opts instrument.TimerOptions,
 ) aggregatorAddUntimedMetrics {
+	var (
+		leaderScope = scope.Tagged(map[string]string{
+			"role": "leader",
+		})
+		nonLeaderScope = scope.Tagged(map[string]string{
+			"role": "non-leader",
+		})
+	)
 	return aggregatorAddUntimedMetrics{
-		aggregatorAddMetricMetrics: newAggregatorAddMetricMetrics(scope, opts),
-		invalidMetricTypes: scope.Tagged(map[string]string{
-			"reason": "invalid-metric-types",
-		}).Counter("errors"),
+		aggregatorAddMetricSuccessMetrics: newAggregatorAddMetricSuccessMetrics(scope, opts),
+
+		leaderErrors:    newAggregatorAddUntimedErrorMetrics(leaderScope),
+		nonLeaderErrors: newAggregatorAddUntimedErrorMetrics(nonLeaderScope),
```

Comment:
Nit: I suggest reducing the boilerplate here (and in other places) by moving to something like this:
```suggestion
		leaderErrors:    newAggregatorAddUntimedErrorMetrics(withRole("leader", scope)),
		nonLeaderErrors: newAggregatorAddUntimedErrorMetrics(withRole("non-leader", scope)),
```

Reply:
https://github.com/m3db/m3/pull/3559/commits/6d01251ee123ea551f405228b99f8937b074f838